k8s-operator/apis/v1alpha1/types_connector.go

```diff
@@ -76,9 +76,19 @@ type ConnectorSpec struct {
 	// Connector node. If unset, hostname defaults to <connector
```

Comment:
After playing around with some invalid combinations, I think it would be nice to surface the hostname/replicas error a bit earlier, I think something like this should work on the `ConnectorSpec` struct:

```go
// +kubebuilder:validation:XValidation:rule="!(has(self.hostname) && has(self.replicas) && self.replicas > 1)",message="The hostname field cannot be specified when replicas is greater than 1."
```

And similarly for hostname/hostnamePrefix:

```go
// +kubebuilder:validation:XValidation:rule="!(has(self.hostname) && has(self.hostnamePrefix))",message="The hostname and hostnamePrefix fields are mutually exclusive."
```